selfdrive/car/hyundai/carcontroller.py

```diff
@@ -102,20 +102,22 @@ def update(self, CC, CS, now_nanos):
       if self.CP.flags & HyundaiFlags.ENABLE_BLINKERS:
         can_sends.append([0x7b1, 0, b"\x02\x3E\x80\x00\x00\x00\x00\x00", self.CAN.ECAN])
 
+    can_canfd_hda2 = self.CP.flags & HyundaiFlags.CAN_CANFD_HDA2
+
     # CAN-FD platforms
-    if self.CP.carFingerprint in CANFD_CAR:
+    if self.CP.carFingerprint in CANFD_CAR or can_canfd_hda2:
       hda2 = self.CP.flags & HyundaiFlags.CANFD_HDA2
       hda2_long = hda2 and self.CP.openpilotLongitudinalControl
 
       # steering control
       can_sends.extend(hyundaicanfd.create_steering_messages(self.packer, self.CP, self.CAN, CC.enabled, apply_steer_req, apply_steer))
 
       # disable LFA on HDA2
-      if self.frame % 5 == 0 and hda2:
+      if self.frame % 5 == 0 and (hda2 or can_canfd_hda2):
         can_sends.append(hyundaicanfd.create_cam_0x2a4(self.packer, self.CAN, CS.cam_0x2a4))
 
       # LFA and HDA icons
-      if self.frame % 5 == 0 and (not hda2 or hda2_long):
+      if self.frame % 5 == 0 and (not hda2 or hda2_long) and not can_canfd_hda2:
         can_sends.append(hyundaicanfd.create_lfahda_cluster(self.packer, self.CAN, CC.enabled))
```

Comment:
Since the ADAS messages seem to be mostly shared with CAN FD, let's just consider it one.